cmake/ImportBSON.cmake

```diff
@@ -0,0 +1,170 @@
+   # extra-alignment enabled. We want to match that setting as our default, for convenience
+   # purposes only.
+   find_path(SYSTEM_BSON_INCLUDE_DIR bson/bson.h PATH_SUFFIXES libbson-1.0)
+   if (SYSTEM_BSON_INCLUDE_DIR AND NOT DEFINED ENABLE_EXTRA_ALIGNMENT)
+      set(CMAKE_REQUIRED_INCLUDES "${SYSTEM_BSON_INCLUDE_DIR}")
+      set (_extra_alignment_default OFF)
+      check_c_source_compiles([[
+         #include <bson/bson.h>
+
+         int main() { }
+      ]] HAVE_SYSTEM_LIBBSON)
+
+      if (HAVE_SYSTEM_LIBBSON)
+         # We have a libbson, check for extra alignment
+         check_c_source_compiles([[
+            #include <bson/bson.h>
+
+            #ifndef BSON_EXTRA_ALIGN
+            #error "Not extra-aligned"
+            #endif
+
+            int main() {}
+         ]] SYSTEM_LIBBSON_IS_EXTRA_ALIGNED)
+         if (SYSTEM_LIBBSON_IS_EXTRA_ALIGNED)
+            # Extra aligned! We'll use extra alignment by default.
+            set (_extra_alignment_default ON)
+         endif ()
+      endif ()
+   endif ()
+cmake_pop_check_state()
+
+set (init OFF)
+if (DEFINED ENABLED_SHARED_BSON)
+   message (STATUS "ENABLE_SHARED_BSON is now named USE_SHARED_LIBBSON")
+   set (init "${ENABLE_SHARED_BSON}")
+endif ()
+option (USE_SHARED_LIBBSON "Dynamically link libbson for the libmongocrypt dynamic library (default is static)" ${init})
+
+if (NOT DEFINED MONGOCRYPT_MONGOC_DIR)
+   # The user did not provide a MONGOCRYPT_MONGOC_DIR, so we'll get one
+   include (FetchContent OPTIONAL)
+   if (NOT COMMAND FetchContent_Declare)
+      # We need FetchContent in order to download the project.
+      message (FATAL_ERROR
+            "No MONGOCRYPT_MONGOC_DIR setting was defined, and the FetchContent.cmake "
+            "module is not available. Upgrade your CMake version, or provide a "
+            "MONGOCRYPT_MONGOC_DIR path to a mongo-c-driver directory (This is required "
+            "for libmongocrypt to find a libbson to use and link against).")
+   endif ()
+   include (FetchMongoC)
+   # The FetchMongoC module defines a MONGOCRYPT_MONGOC_DIR for use to use
+endif ()
+
+message (STATUS "Using [${MONGOCRYPT_MONGOC_DIR}] as a sub-project for libbson")
+
+function (_import_bson_add_subdir)
+   # Disable AWS_AUTH, to prevent it from building the kms-message symbols, which we build ourselves
+   set (ENABLE_MONGODB_AWS_AUTH OFF CACHE BOOL "Disable kms-message content in mongoc for libmongocrypt" FORCE)
+   # Disable install() for the libbson static library. We'll do it ourselves
+   set (ENABLE_STATIC BUILD_ONLY)
+   # Disable over-alignment of bson types
+   set (ENABLE_EXTRA_ALIGNMENT ${_extra_alignment_default} CACHE BOOL "Toggle extra alignment of bson_t")
+   # Add the subdirectory as a project. EXCLUDE_FROM_ALL to inhibit building and installing of components unless requested
+   add_subdirectory ("${MONGOCRYPT_MONGOC_DIR}" _mongo-c-driver EXCLUDE_FROM_ALL)
```

Comment:
I remember seeing this issue previously. The mongo-c project overrides the message() command to be a no-op, for some reason. This was removed in a later version, but remains in 1.17.0.

Reply:
Confirmed. It seems like this issue has been resolved since [C Driver release version 1.17.3](https://github.com/mongodb/mongo-c-driver/compare/1.17.2...1.17.3). Good to know.

Is there a reason the C Driver version is currently being pinned to 1.17.0 and not a newer release?

Reply:
No specific reason other than "it already was". There are some platform issues regarding the newer Mongo-c versions because of atomics, but bumping to a bugfix of 1.17 should be fine.

Reply:
Wow, that's a throwback.  Yes, it was a rather unfortunate part of the build.  I also see no issue to bumping to the latest 1.17 (which happens to be 1.17.7).